src/contracts/actions-builders/EverclearTokenStake.sol

```diff
@@ -0,0 +1,151 @@
+import {IEverclearTokenStake} from 'interfaces/actions-builders/IEverclearTokenStake.sol';
+import {IGateway} from 'interfaces/external/IGateway.sol';
+import {ISpokeBridge} from 'interfaces/external/ISpokeBridge.sol';
+import {IVestingEscrow} from 'interfaces/external/IVestingEscrow.sol';
+import {IVestingWallet} from 'interfaces/external/IVestingWallet.sol';
+import {IxERC20Lockbox} from 'interfaces/external/IxERC20Lockbox.sol';
+
+/**
+ * @title EverclearTokenStake
+ * @notice Contract that increases the stake of CLEAR
+ */
+contract EverclearTokenStake is IEverclearTokenStake {
+  // ~~~ STORAGE ~~~
+
+  /// @inheritdoc IEverclearTokenStake
+  IVestingEscrow public immutable VESTING_ESCROW;
+
+  /// @inheritdoc IEverclearTokenStake
+  IVestingWallet public immutable VESTING_WALLET;
+
+  /// @inheritdoc IEverclearTokenStake
+  ISpokeBridge public immutable SPOKE_BRIDGE;
+
+  /// @inheritdoc IEverclearTokenStake
+  IxERC20Lockbox public immutable CLEAR_LOCKBOX;
+
+  /// @inheritdoc IEverclearTokenStake
+  IERC20 public immutable NEXT;
+
+  /// @inheritdoc IEverclearTokenStake
+  IERC20 public immutable CLEAR;
+
+  /// @inheritdoc IEverclearTokenStake
+  uint256 public immutable LOCK_TIME;
+
+  // ~~~ CONSTRUCTOR ~~~
+
+  /**
+   * @notice Constructor that sets up the variables
+   * @param _vestingEscrow The vesting escrow contract address
+   * @param _vestingWallet The vesting wallet contract address
+   * @param _spokeBridge The spoke bridge contract address
+   * @param _clearLockbox The clear lockbox contract address
+   * @param _next The NEXT contract address
+   * @param _clear The CLEAR contract address
+   * @param _lockTime The lock time
+   */
+  constructor(
+    address _vestingEscrow,
+    address _vestingWallet,
+    address _spokeBridge,
+    address _clearLockbox,
+    address _next,
+    address _clear,
+    uint256 _lockTime
+  ) {
+    VESTING_ESCROW = IVestingEscrow(_vestingEscrow);
+    VESTING_WALLET = IVestingWallet(_vestingWallet);
+    SPOKE_BRIDGE = ISpokeBridge(_spokeBridge);
+    CLEAR_LOCKBOX = IxERC20Lockbox(_clearLockbox);
+    NEXT = IERC20(_next);
+    CLEAR = IERC20(_clear);
+    LOCK_TIME = _lockTime;
+  }
+
+  // ~~~ ACTIONS METHODS ~~~
+
+  /// @inheritdoc IActionsBuilder
+  function getActions() external view returns (Action[] memory _actions) {
+    _actions = new Action[](6);
+
+    // NOTE: since this is a view function and does not update state, we need to calculate
+    // how much tokens are transferred when calling release() on step 2.
+
+    // First get the unclaimed amount
+    uint256 _unclaimed = VESTING_ESCROW.unclaimed();
+    // Get current balance of NEXT
+    uint256 _nextBalance = NEXT.balanceOf(address(VESTING_WALLET));
+    // Get the releaseable amount (same behaviour as releaseable() on VESTING_WALLET)
+    uint256 _amountReleasable = VESTING_WALLET.vestedAmount(uint64(block.timestamp)) - VESTING_WALLET.released();
```

Comment:
i'd add a note saying that we're getting "the releaseable amount at the time of enqueuing"